cmd/ocsp-responder/main.go

```diff
@@ -223,104 +245,132 @@ func (src *dbSource) Response(ctx context.Context, req *ocsp.Request) ([]byte, h
 
 	// If the primary source returns first, check the output and return
 	// it. If the secondary source wins, then wait for the primary so the
-	// results from the seconary can be verified. It is important that we
+	// results from the secondary can be verified. It is important that we
 	// never return a response from the redis source that is good if mysql
 	// has a revoked status. If the secondary source wins the race and
-	// passes these checks, return it's response instead.
+	// passes these checks, return its response instead.
 	select {
 	case <-ctx.Done():
 		err := fmt.Errorf("looking up OCSP response for serial: %s err: %w", serialString, ctx.Err())
-		src.log.Debugf(err.Error())
-		ocspLookups.WithLabelValues("canceled").Inc()
+		src.log.Debugf("%s", err)
+		src.ocspLookups.WithLabelValues("canceled").Inc()
 		return nil, nil, err
 	case primaryResult := <-primaryChan:
 		if primaryResult.err != nil {
-			src.log.AuditErrf("Looking up OCSP response: %s", err)
-			ocspLookups.WithLabelValues("mysql_failed").Inc()
-			sourceUsed.WithLabelValues("error_returned").Inc()
+			src.log.AuditErrf("Looking up OCSP response in DB: %s", err)
+			src.ocspLookups.WithLabelValues("mysql_failed").Inc()
+			src.sourceUsed.WithLabelValues("error_returned").Inc()
```

Comment:
It looks like `error_returned` is meant to be incremented any time an error is returned from this function. Similar to the logging issue I mentioned, there's code at the level above this one that serves the same purpose: https://github.com/letsencrypt/boulder/blob/main/ocsp/responder.go#L372. I think that does a more thorough job because it's guaranteed to catch all error returns.

Reply:
We talked about this in a call. The error_returned label is a third option for source used. It will either be "redis", "mysql", or "error_returned" and makes sense to leave this in for this case to account for and track all possibilities.